__TESTS__/unit/actions/Overlay.test.ts

```diff
@@ -0,0 +1,30 @@
+import {fill} from "../../../src/actions/resize/Resize";
+import {auto, best} from "../../../src/actions/quality/Quality";
+import {imageLayer} from "../../../src/actions/layers/overlay";
+import {imageSource} from "../../../src/actions/layers/Sources";
+import TransformableImage from "../../../src/transformation/TransformableImage";
+
+describe('Tests for overlay actions', () => {
+  it('Parses an overlay with an imageSource', () => {
+    const tImage = new TransformableImage('sample');
+    tImage
+      .setConfig({
+        cloud: {
+          cloudName:'demo'
+        }
+      })
+      .resize(fill(200, 200))
+      .quality(best())
+      .overlay(
+        // imageLayerAction is what registers on the transformation(which has toString() method)
+        // imageSource is the parameter for imageLayerAction
+        imageLayer( imageSource('sample')
+          .resize(fill(50, 50))
+          .quality(auto())
```

Comment:
That is the least important thing in this review, however it's still an inaccuracy - being transformable, layers to support format/quality actions (since they support all actions), however it's quite meaningless to use these actions on a layer. So while not really wanting to block it in the code (it's not even blocked on the backend), we do try to have realistic use-cases in our public tests.